src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs

```diff
@@ -0,0 +1,683 @@
+                context.Renew();
+
+                using (context.OpenReadTransaction())
+                {
+                    var list = tss.Stats.GetTimeSeriesByPolicyFromStartDate(context, collectionName, policy.Name, deleteFrom, TimeSeriesRetentionCommand.BatchSize).ToList();
+                    if (list.Count == 0)
+                        return;
+
+                    request.To = deleteFrom;
+                    var cmd = new TimeSeriesRetentionCommand(list, request);
+                    await _database.TxMerger.Enqueue(cmd);
+                }
+            }
+        }
+
+        private void PrepareRollUps(DocumentsOperationContext context, DateTime currentTime, long take, List<RollUpState> states, out Stopwatch duration)
+        {
+            duration = Stopwatch.StartNew();
+
+            var table = context.Transaction.InnerTransaction.OpenTable(RollUpSchema, TimeSeriesRollUps);
+            if (table == null)
+                return;
+
+            var currentTicks = currentTime.Ticks;
+
+            foreach (var item in table.SeekForwardFrom(RollUpSchema.Indexes[NextRollUpIndex], Slices.BeforeAllKeys, 0))
+            {
+                if (take <= 0)
+                    return;
+
+                var rollUpTime = DocumentsStorage.TableValueToEtag((int)RollUpColumn.NextRollUp, ref item.Result.Reader);
+                if (rollUpTime > currentTicks)
+                    return;
+
+                DocumentsStorage.TableValueToSlice(context, (int)RollUpColumn.Key, ref item.Result.Reader,out var key);
+                SplitKey(key, out var docId, out var name);
+
+                var state = new RollUpState
+                {
+                    Key = key,
+                    DocId = docId,
+                    Name = name,
+                    Collection = DocumentsStorage.TableValueToString(context, (int)RollUpColumn.Collection, ref item.Result.Reader),
+                    NextRollUp = new DateTime(rollUpTime),
+                    RollUpPolicy = DocumentsStorage.TableValueToString(context, (int)RollUpColumn.PolicyToApply, ref item.Result.Reader),
+                    Etag = DocumentsStorage.TableValueToLong((int)RollUpColumn.Etag, ref item.Result.Reader),
+                    ChangeVector = DocumentsStorage.TableValueToChangeVector(context, (int)RollUpColumn.ChangeVector, ref item.Result.Reader)
+                };
+
+                states.Add(state);
+                take--;
+            }
+        }
+
+        public static void SplitKey(Slice key, out string docId, out string name)
+        {
+            var separatorIndex = key.Content.IndexOf(SpecialChars.RecordSeparator);
+            docId = key.Content.Substring(separatorIndex);
+            name = key.Content.Substring(separatorIndex + 1, key.Content.Length - separatorIndex - 1);
+        }
+
+        internal class TimeSeriesRetentionCommand : TransactionOperationsMerger.MergedTransactionCommand
+        {
+            public const int BatchSize = 1024;
```

Comment:
What is this used for?

Reply:
Used elsewhere to limit how many items we scan per batch